addons/event/tools/esc_label_tools.py

```diff
@@ -0,0 +1,417 @@
+            color = self._parse_color_string(color)
+        if isinstance(bg_color, str):
+            bg_color = self._parse_color_string(bg_color)
+
+        self._command += f"^F(C{color[0]},{color[1]},{color[2]},{alpha},D,{bg_color[0]},{bg_color[1]},{bg_color[2]},{bg_alpha},D"
+        return self
+
+    def print_complete(self):
+        """
+        Signal that the current label is complete.
+        """
+        self._command += "^S(CUB,S,L"
+        return self
+
+    def save_canvas(self):
+        """
+        Save the current print as a template.
+        """
+        self._command += "^C(SN"
+        return self
+
+    def load_canvas(self):
+        """
+        Load the template that was saved earlier with `save_canvas`.
+        """
+        self._command += "^C(L"
+        return self
+
+
+def setup_printer(layout: dict):
+    return (EscLabelCommand()
+        .delete_files("*.*")  # Clean-up printer memory
+        .set_resolution(600)  # 600 DPI
+        .set_media_coating("M1")  # Matte paper
+        .set_media_type("DL")  # Die-cut label
+        .set_media_shape("FP")  # Fanfold paper
+        .set_media_source("ER")  # External paper feed
+        .set_edge_detection("W")  # Detect gap between labels
+        .set_print_quality("N")  # Normal quality
+        .set_printable_area(layout["print_width"],
+                            layout["print_height"] * 2 if layout["double_sided"] else layout["print_height"])
+        .set_label_gap(layout["label_gap"])
+        .set_left_gap(48)
+        .set_printing_offset(layout["print_offset_left"], layout["print_offset_top"])
+        .wrap_command()
+    )
+
+
+def print_centered_text(layout: dict, text: str, y_position: int, font_size: int, command: EscLabelCommand, flip=False):
+    if flip:
+        y_position = layout["print_height"] * 2 - y_position - font_size
+    command.print_text(
+        text,
+        position=(layout["text_margin"], y_position),
+        font_size=(font_size, font_size),
+        wrap_width=layout["print_width"] - (layout["text_margin"] * 2),
+        align="C",  # Center aligned
+        rotation="I" if flip else "N"  # (I)nverted if printing flipped, otherwise (N)ormal
+    )
+
+
+layout_96x82 = {
+    "print_width": 2340,
+    "print_height": 1965,
+    "print_offset_left": -30,
+    "print_offset_top": 50,
+    "double_sided": True,
+    "label_gap": 80,
+    "text_margin": 160,
+    "logo_width": 800,
+    "logo_height": 300,
+    "ticket_bg_height": 275,
+    "event_name_font_size": 150,
+    "details_font_size": 80,
+    "attendee_name_font_size": 130,
+    "company_font_size": 100,
+    "ticket_font_size": 110,
+    "event_name_y_pos": 350,
+    "date_y_pos": 550,
+    "address_y_pos": 670,
+    "attendee_name_y_pos": 825,
+    "company_y_pos": 980,
+    "answers_y_pos": 1120,
+    "logo_y_pos": 1200,
+    "ticket_text_y_pos": 1780,
+    "custom_text_y_pos": 1550,
+    "secondary_text_color": "#374151",
+    "secondary_text_alpha": 200
+}
```

Comment:
Change from previous PR: Layout parameters are hard-coded here rather than as a new model, so they can't be changed by the end user.
We could still improve to allow more configuration in `master`